05-http-server-streams/01-file-server-get/server.js

```diff
@@ -1,22 +1,52 @@
 const url = require('url');
 const http = require('http');
 const path = require('path');
+const fs = require('fs');
 
 const server = new http.Server();
 
+const sendFile = (res, filepath) => {
+  const readStream = fs.createReadStream(filepath);
+
+  readStream.on('open', () => {
+    readStream.pipe(res);
+  });
+
+  readStream.on('error', (err) => {
+    res.statusCode = 500;
+    res.end(err);
+  });
+};
+
 server.on('request', (req, res) => {
   const pathname = url.parse(req.url).pathname.slice(1);
-
   const filepath = path.join(__dirname, 'files', pathname);
 
   switch (req.method) {
-    case 'GET':
+    case 'GET': {
+      if (pathname.includes('/')) {
+        res.statusCode = 400;
+        res.end('Subfolders are not supported');
+        return;
```

Comment:
Is there any point in doing a return?

Reply:
Sure. Otherwise, the execution of the function will continue and will try to open the file. 
Personally I prefer the [`early return` pattern](https://medium.com/swlh/return-early-pattern-3d18a41bba8)